apollo-router/src/configuration/cors.rs

```diff
@@ -1,28 +1,159 @@
 //! Cross Origin Resource Sharing (CORS configuration)
+//!
+//! This module provides configuration structures for CORS (Cross-Origin Resource Sharing) settings.
+//!
+//! # Default Behavior
+//!
+//! When the `policies` field is omitted from the CORS config, the router uses a default policy:
+//! - **Origins:** `["https://studio.apollographql.com"]`
+//! - **Methods:** `["GET", "POST", "OPTIONS"]`
+//! - **Allow credentials:** `false`
+//! - **Allow any origin:** `false`
+//!
+//! # Policy Configuration
+//!
+//! When specifying individual policies within the `policies` array:
+//! - **Origins:** Defaults to `["https://studio.apollographql.com"]` when no policies are specified at all.
+//!   If you specify any policies, you must explicitly set origins for each policy.
+//! - **Match origins:** Defaults to an empty list (no regex matching) unless explicitly set
+//! - **Allow credentials:** Has three possible states:
+//!   - not specified: Use the global default allow_credentials
+//!   - `true`: Enable credentials for this policy
+//!   - `false`: Disable credentials for this policy
+//! - **Allow headers:** Inherits global headers if empty, otherwise uses policy-specific headers
+//! - **Expose headers:** Inherits global headers if empty, otherwise uses policy-specific headers
```

Comment:
For these two (allow headers and expose headers), from the perspective of a customer crafting their config.yaml.... I think there are the same 3 states that can be done in the yaml file.  
- not specified - use the global setting
- empty array - use the global setting
- an array with policy specific headers - use these policy-specific behavior